sdk/storage/azdatalake/directory/client.go

```diff
@@ -174,65 +185,189 @@ func (d *Client) BlobURL() string {
 
 //TODO: create method to get file client - this will require block blob to have a method to get another block blob
 
-// Create creates a new directory (dfs1).
+// Create creates a new file (dfs1).
 func (d *Client) Create(ctx context.Context, options *CreateOptions) (CreateResponse, error) {
-	return CreateResponse{}, nil
+	lac, mac, httpHeaders, createOpts, cpkOpts := options.format()
+	resp, err := d.generatedDirClientWithDFS().Create(ctx, createOpts, httpHeaders, lac, mac, nil, cpkOpts)
+	err = exported.ConvertToDFSError(err)
+	return resp, err
 }
 
-// Delete removes the directory (dfs1).
+// Delete deletes directory and any path under it (dfs1).
 func (d *Client) Delete(ctx context.Context, options *DeleteOptions) (DeleteResponse, error) {
-	//TODO: pass recursive = true
-	return DeleteResponse{}, nil
+	lac, mac, deleteOpts := path.FormatDeleteOptions(options, true)
+	resp, err := d.generatedDirClientWithDFS().Delete(ctx, deleteOpts, lac, mac)
+	err = exported.ConvertToDFSError(err)
+	return resp, err
 }
 
-// GetProperties returns the properties of the directory (blob3). #TODO: we may just move this method to path client
+// GetProperties gets the properties of a file (blob3)
 func (d *Client) GetProperties(ctx context.Context, options *GetPropertiesOptions) (GetPropertiesResponse, error) {
-	// TODO: format blob response to path response
-	return GetPropertiesResponse{}, nil
+	opts := path.FormatGetPropertiesOptions(options)
+	var respFromCtx *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &respFromCtx)
+	resp, err := d.blobClient().GetProperties(ctxWithResp, opts)
+	newResp := path.FormatGetPropertiesResponse(&resp, respFromCtx)
+	err = exported.ConvertToDFSError(err)
+	return newResp, err
+}
+
+func (d *Client) renamePathInURL(newName string) (string, string, string) {
```

Comment:
can we make sure to add individual UT for this method?

Reply:
absolutely will put as TODO